### PR TITLE
[KRYS-3056] Add Klay chain

### PR DESCRIPTION
**Pull request by MrQKyber**

### CODE REVIEW

File: KyberNetwork/Settings/Types/AllChains.swift [hunk 1]
```diff
@@ -287,4 +287,24 @@ public struct AllChains {
     quoteToken: "SOL",
     apiChainPath: "solana"
   )
+  
+  public static let klaytnMainnetRPC = CustomRPC (
+    chainID: 8217,
+    type: " KIP-7",
```
Comment (tungnguyen20) [inline KyberNetwork/Settings/Types/AllChains.swift:293]: Please remove redundant space

File: KyberNetwork/Settings/Types/AllChains.swift [hunk 2]
```diff
@@ -287,4 +287,24 @@ public struct AllChains {
...
+    endpointKyber: "https://public-node-api.klaytnapi.com/v1/cypress",
+    endpointAlchemy: "https://public-node-api.klaytnapi.com/v1/cypress",
+    etherScanEndpoint: "https://scope.klaytn.com/",
+    webScanName: "KlaytnScope",
+    ensAddress: "",
+    wrappedAddress: "",
+    apiEtherscanEndpoint: "",
+    proxyAddress: "0x864F01c5E46b0712643B956BcA607bF883e0dbC5",
+    quoteTokenAddress: "0x089fa00784cca541409947e9bd098ae816d00822",
+    chainIcon: "chain_aurora_icon",
```
Comment (tungnguyen20) [inline KyberNetwork/Settings/Types/AllChains.swift:306]: chain_aurora_icon

File: KyberNetwork/KyberNetwork/Coordinators/BalanceCoordinator/KNLoadBalanceCoordinator.swift [hunk 3]
```diff
@@ -429,7 +429,7 @@ class KNLoadBalanceCoordinator {
   }
 
   func loadLendingBalances(forceSync: Bool = false, completion: @escaping (Bool) -> Void) {
-    guard KNGeneralProvider.shared.currentChain != .solana else { return }
+    guard KNGeneralProvider.shared.currentChain != .solana || KNGeneralProvider.shared.currentChain != .klaytn else { return }
```
Comment (tungnguyen20) [inline KyberNetwork/KyberNetwork/Coordinators/BalanceCoordinator/KNLoadBalanceCoordinator.swift:432]: Please create an extension for `ChainType` so this line can be refactored to
`
guard KNGeneralProvider.shared.currentChain.isLendingBalancesSupported else { return }
`

File: KyberNetwork/KyberNetwork/Screens/Tabbars/Invest/ViewModels/ExploreViewModel.swift [hunk 4]
```diff
@@ -55,7 +55,7 @@ class ExploreViewModel {
     menuItems.append(.swap)
     menuItems.append(.transfer)
     
-    if KNGeneralProvider.shared.currentChain != .solana {
+    if KNGeneralProvider.shared.currentChain != .solana || KNGeneralProvider.shared.currentChain != .klaytn {
```
Comment (tungnguyen20) [inline KyberNetwork/KyberNetwork/Screens/Tabbars/Invest/ViewModels/ExploreViewModel.swift:58]: same to this

**Review by toidv — APPROVED** [review]

File: KyberNetwork/KyberNetwork/Screens/Tabbars/Invest/ViewModels/ExploreViewModel.swift [hunk 5]
```diff
@@ -55,7 +55,7 @@ class ExploreViewModel {
     menuItems.append(.swap)
     menuItems.append(.transfer)
     
-    if KNGeneralProvider.shared.currentChain != .solana {
+    if KNGeneralProvider.shared.currentChain.isSupportSwap() {
```
Comment (tungnguyen20) [inline KyberNetwork/KyberNetwork/Screens/Tabbars/Invest/ViewModels/ExploreViewModel.swift:58]: `isSupportSwap` is not correct here

File: KyberNetwork/KyberNetwork/Coordinators/BalanceCoordinator/KNLoadBalanceCoordinator.swift [hunk 6]
```diff
@@ -429,7 +429,7 @@ class KNLoadBalanceCoordinator {
   }
 
   func loadLendingBalances(forceSync: Bool = false, completion: @escaping (Bool) -> Void) {
-    guard KNGeneralProvider.shared.currentChain != .solana else { return }
+    guard KNGeneralProvider.shared.currentChain.isSupportSwap() else { return }
```
Comment (tungnguyen20) [inline KyberNetwork/KyberNetwork/Coordinators/BalanceCoordinator/KNLoadBalanceCoordinator.swift:432]: `isSupportSwap` is not correct here

File: KyberNetwork/KyberNetwork/Screens/Tabbars/KyberSwap/NewDesign/ViewModel/KSwapViewModel.swift [hunk 7]
```diff
@@ -879,7 +879,7 @@ class KSwapViewModel {
   }
   
   var shouldShowCommingSoon: Bool {
-    return KNGeneralProvider.shared.currentChain == .solana
+    return KNGeneralProvider.shared.currentChain == .solana || KNGeneralProvider.shared.currentChain == .klaytn
```
Comment (tungnguyen20) [inline KyberNetwork/KyberNetwork/Screens/Tabbars/KyberSwap/NewDesign/ViewModel/KSwapViewModel.swift:882]: `!KNGeneralProvider.shared.currentChain.isSupportSwap()`

**Pull request merged**